runtime/vm/createramclass.cpp

```diff
@@ -1995,6 +1995,13 @@ internalCreateRAMClassDone(J9VMThread *vmThread, J9ClassLoader *classLoader, J9R
 			}
 		}
 
+		/* link classes in segment */
+		if (NULL != segment) {
+			/* segment is NULL in failure paths */
+			state->ramClass->nextClassInSegment = *(J9Class **) segment->heapBase;
+			*(J9Class **)segment->heapBase = state->ramClass;
```

Comment:
With this change, is the code above that handles the "already loaded" case still valid / required?  I think it can be removed now as there's no reason to link the abandoned class into anything as it won't be part of the list until this point

https://github.com/eclipse/openj9/blob/384c85680c1015e456521b6ab159ad00accf8684/runtime/vm/createramclass.cpp#L1974-L1980